### PR TITLE
[Source] Fix a crash when searching for compatible versions

### PR DESCRIPTION
**Pull request by amorde**
Closes https://github.com/CocoaPods/Core/issues/482

- [x] Fix Rubocop violations

### CODE REVIEW

File: lib/cocoapods-core/source/metadata.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def path_fragment(pod_name, version = nil)
       def last_compatible_version(target_version)
         return unless minimum_cocoapods_version
         return if minimum_cocoapods_version <= target_version
-        @last_compatible_versions.reverse_each.bsearch { |v| v <= target_version }.tap do |version|
+        Array(@last_compatible_versions.reverse_each).bsearch { |v| v <= target_version }.tap do |version|
```
Comment (segiddins) [inline lib/cocoapods-core/source/metadata.rb]: this could be simplified as `@last_compatible_versions.reverse.bsearch`

Comment (amorde) [reply lib/cocoapods-core/source/metadata.rb]: oo that's much better. will do

**Comment by dnkoutso** [comment]
@amorde needs rebase and comment update to land

**Comment by amorde** [comment]
Yup will try to get this in today

**Comment by dnkoutso** [comment]
Quick ping on this would love to land it.

**Comment by dnkoutso** [comment]
Also can be in `1-7-stable` for 1.7.x release

**Pull request merged**